site/_posts/2020-09-03-oidc-auth-api-usage.md

```diff
@@ -0,0 +1,173 @@
+
+   # Step 3 (OPTIONAL) Use the access_token to do Introspection
+   curl -k -L --user ${oidc_client_id}:${oidc_client_secret} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "token=${access_token}" ${oidc_auth_introspection_endpoint}
+
+
+   # Step 4: Accessing MiQ API using the access_token in the header
+   curl -L -vvv -k -X GET -H "Authorization: Bearer ${access_token}" https://${miq_server_name}/api/users | jq
+```
+
+
+## Example: Using a ManageIQ API Auth Token
+---------------------------------------------------------------------
+
+This example details the steps for using a ManageIQ API Auth Token for a non-admin user.
+
+The steps are:
+1. Request the JWT Token from the OpenID-Connect provider
+1. Retrieve the access_token from the JWT
+1. (OPTIONAL) Use the access_token to do Introspection
+1. Request an MiQ API Authentication Token using the access_token in the header:
+1. Retrieve the API authentication token from the result
+1. Accessing MiQ API Using the MiQ API Auth Token
+
+The bash example commands:
+
+```bash
+# Accessing the ManageIQ API using a ManageIQ API Auth Token
+
+   # Step 1: Request the JWT Token from the OpenID-Connect provider
+   jwt_token=$(curl -k -L --user ${user}:${password} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "grant_type=password" -d "client_id=${oidc_client_id}" -d "client_secret=${oidc_client_secret}" -d "username=${user}" -d "password=${password}" ${oidc_provider_token_endpoint} )
+
+   # Step 2: Retrieve the access_token from the JWT
+   access_token=$(echo $jwt_token | jq -r '.access_token')
+
+   # Step 3: (OPTIONAL) Use the access_token to do Introspection
+   curl -k -L --user ${oidc_client_id}:${oidc_client_secret} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "token=${access_token}" ${oidc_auth_introspection_endpoint}
+
+   # Step 4: Request an MiQ API Authentication Token:
+   result=$(curl -L -vvv -k -X GET -H "Authorization: Bearer ${access_token}" https://${miq_server_name}/api/auth)
+
+   # Step 5: Retrieve the API authentication token from the result
+   api_auth_token=`echo $result | jq -r '.auth_token'`
+
+   # Step 6: Accessing MiQ API Using the MiQ API Auth Token:
+   curl -L -vvv -k -X GET  -H "Accept: application/json" -H "X-Auth-Token: ${api_auth_token}" https://${miq_server_name}/api/users | jq
+
+```
+
+## Example: Using Basic Auth For Admin
+---------------------------------------------------------------------
+
+The ManageIQ OpenID-Connect is configured to treat the admin user as a special case.
+The admin user is not authenticated by the OpenID-Connect provider. The admin user is
+the only mode where basic authentication is supported.
+
+The bash example command:
+
+```bash
+# Accessing the MiQ API Using basic admin:password
+
+  # Using basic authentication for the admin user.
+   curl -L -vvv -k --user admin:${admin_password} -X GET -H 'Accept: application/json' https://${miq_server_name}/api/users | jq
+```
```

Comment:
is this needed here since we talk about it in the API doc ?   I guess you can keep but  mention they can  reference https://www.manageiq.org/docs/reference/latest/api/overview/auth.html specifically here for examples on basic and token based auth in the API.